runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/config/CompressionClientConfig.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.http.config
+
+import aws.smithy.kotlin.runtime.http.compression.CompressionAlgorithm
+import aws.smithy.kotlin.runtime.http.compression.Gzip
+
+@DslMarker
+public annotation class CompressionClientConfigDsl
+
+public interface CompressionClientConfig {
+    public val requestCompression: RequestCompressionConfig
+
+    public interface Builder {
+        public fun requestCompression(block: RequestCompressionConfig.Builder?.() -> Unit) {
```

Comment:
**correctness**: We shouldn't be passing a nullable type for this, it's not ergonomic for users